resource_activity/models/resource_available.py

```diff
@@ -12,9 +12,14 @@ class ResourceAvailable(models.Model):
     _description = "Resource Available"
 
     name = fields.Char(related="resource_id.serial_number", string="Name")
-    resource_id = fields.Many2one("resource.resource", string="Resource", required=True)
+    resource_id = fields.Many2one(
+        comodel_name="resource.resource", string="Resource", required=True
+    )
```

Comment:
this should be a stored (settable) computed field that depends on `allocation_id` and `allocation_id.resource_id`. it would allow to automatically update the `resource_id` when it is change in the allocation.

same remark for the `state` field.